fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -838,4 +850,50 @@ public ChecksumIndexInput openChecksumInput(final String name, final IOContext c
     Cache<Pair<Long, Integer>, CompletableFuture<byte[]>> getBlockCache() {
         return blockCache;
     }
+
+    /**
+     * Get a primary key segment index if enabled.
+     * @return index or {@code null} if not enabled
+     */
+    @Nullable
+    public LucenePrimaryKeySegmentIndex getPrimaryKeySegmentIndex() {
+        if (!primaryKeySegmentIndexEnabled) {
+            return null;
+        }
+        if (primaryKeySegmentIndex == null) {
```

Comment:
Does this need to be `synchronized`? I'm fairly sure that merging uses a bunch of threads.
`LucenePrimaryKeySegmentIndex` doesn't have any mutable state, so it's probably fine to have multiple different instances exist. Probably worth commenting otherwise.

Reply:
I added a `synchronized` around this bit, just be safe. There weren't any already in the class, which maybe indicates other problems._